packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -88,11 +89,15 @@ const getLocationSearch = (req) => {
         const values = Array.isArray(value) ? value : [value]
 
         values.forEach((v) => {
-            newParams.append(key, v)
+            // Because URLSearchParams constructor perceives + as space, but append does not
+            // we want to make sure we run the key-value through new URLSearchParams constructor to parse + as space
+            // before we append it to newParams
+            const temp = new URLSearchParams(`${key}=${v}`)
+            newParams.append(key, temp.get(key))
```

Comment:
```suggestion
            const decoded = decodeURIComponent(v).replace(/\+/g, ' ')
            newParams.append(key, decoded)
```
The constructor expects a URL encoded string, but the set/append methods expect decoded strings, so we can skip the `URLSearchParams` wrapper and just do the decoding ourselves.

ref: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#preserving_plus_signs (although we're doing the opposite)
ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent#decoding_query_parameters_from_a_url

Reply:
Note that whichever approach we choose, this is a breaking change! Some customers may want `+` to be `+`, even if others want `+` to be ` `. The result should be controlled by the end user, and it should default to the existing behavior.

Reply:
I've added a flag to let user control the behavior and added tests for that